keylime/agentstates.py

```diff
@@ -6,28 +6,49 @@
 
 import threading
 
-from keylime.ima_ast import START_HASH, FF_HASH
+from keylime.ima_ast import get_FF_HASH, get_START_HASH
+from keylime.common.algorithms import Hash
 from keylime.ima_file_signatures import ImaKeyrings
 
 class TPMState():
     """ TPMState models the state of the TPM's PCRs """
+
     def __init__(self):
         """ constructor """
         self.pcrs = {}
+        self.hash_alg = {}
         for pcr_num in range(0, 24):
             self.reset_pcr(pcr_num)
 
     def reset_pcr(self, pcr_num):
         """ Reset a specific PCR """
-        if 17 <= pcr_num <= 23:
-            self.pcrs[pcr_num] = FF_HASH
-        else:
-            self.pcrs[pcr_num] = START_HASH
-
-    def get_pcr(self, pcr_num):
-        """ Get the state of a PCR """
+        self.pcrs[pcr_num] = None
+
+    def get_pcr(self, pcr_num, hash_alg: Hash = None):
+        """
+        Get the state of a PCR.
+
+        First call requires to have an hash_alg to be specified and sets the hash algorithm for that PCR.
+
+        :returns: PCR value or None if not initialized and no hash_alg was specified
+        """
+        if pcr_num not in self.hash_alg:
+            if hash_alg:
+                self.hash_alg[pcr_num] = hash_alg
+            else:
+                return None
+
+        if self.pcrs[pcr_num] is None:
```

Comment:
Create new `init_pcr(self, pcr_num)` for this initialization?

Reply:
This is a good idea. Done.